invenio_github/tasks.py

```diff
@@ -114,30 +114,37 @@ def _get_err_obj(msg):
             err['error_id'] = str(g.sentry_event_id)
         return err
 
+    def _release_rest_exception_handler(release, ex):
+        """Handler for RestException."""
+        release.model.errors = json.loads(ex.get_body())
+
+    def _release_gh_metadata_handler(release, ex):
+        """Handler for CustomGithubMetadataError."""
+        release.model.errors = {'errors': ex.message}
+
+    def _release_default_exception_handler(release, ex):
+        """Default handler."""
+        release.model.errors = _get_err_obj('Unknown error occured.')
+
     try:
         release.publish()
         release.model.status = ReleaseStatus.PUBLISHED
-    except RESTException as rest_ex:
-        release.model.errors = json.loads(rest_ex.get_body())
-        release.model.status = ReleaseStatus.FAILED
-        current_app.logger.exception(
-            u'Error while processing {release}'.format(release=release.model))
-    # TODO: We may want to handle GitHub errors differently in the future
-    # except GitHubError as github_ex:
-    #     release.model.errors = {'error': str(e)}
-    #     release.model.status = ReleaseStatus.FAILED
-    #     current_app.logger.exception(
-    #         'Error while processing {release}'
-    #         .format(release=release.model))
-    except CustomGitHubMetadataError as e:
-        release.model.errors = _get_err_obj(str(e))
-        release.model.status = ReleaseStatus.FAILED
-        current_app.logger.exception(
-            u'Error while processing {release}'.format(release=release.model))
-    except Exception:
-        release.model.errors = _get_err_obj('Unknown error occured.')
+
+    except Exception as ex:
+        default_error_handlers = [
```

Comment:
This definition can be moved outside the function as well and since it's a constant it should also be in all upper-case